### PR TITLE
Possibility to add instance_id and omit token

### PR DESCRIPTION
**Pull request by Jax-p**
Based on [PR and its review](https://github.com/Unleash/unleash-client-nextjs/pull/26) I've created new PR with different approach. It still makes it work with GitLab FF but instead of removing `defaultToken` it adds possibility to set token to `false` and omit it and also adds possibility to set instance_id. So this solution does not edit any of existing tests.

- InstanceID works like a token so I find it somehow sensitive. Because of that it masks output in debug and dumps only first and last 2 characters.
- Token can be omitted by setting up env value to 'false'

Closes https://github.com/Unleash/unleash-client-nextjs/issues/21

### CODE REVIEW

File: lib/src/getDefinitions.ts [hunk 1]
```diff
@@ -42,14 +52,23 @@ export const getDefinitions = async (
...
-  const response = await fetch(url, {
-    ...fetchOptions,
+  const fetchUrl = new URL(url);
+
+  if (token) {
+    Object.assign(headers, { Authorization: token });
+  }
+
+  if (instanceId) {
+    fetchUrl.searchParams.append('instance_id', instanceId)
```
Comment (Tymek) [inline lib/src/getDefinitions.ts:67]: 👍🏻

File: lib/src/getDefinitions.ts [hunk 2]
```diff
@@ -8,13 +8,17 @@ export const getDefaultConfig = (defaultAppName = "nextjs") => {
     process.env.UNLEASH_SERVER_API_URL ||
     process.env.NEXT_PUBLIC_UNLEASH_SERVER_API_URL;
 
+  const token = process.env.UNLEASH_SERVER_API_TOKEN;
+  const resolvedToken: string | false = (token === 'false' ? false : token || defaultToken);
```
Comment (Tymek) [inline lib/src/getDefinitions.ts:14]: Is there a situation where you'd use both token and `UNLEASH_SERVER_INSTANCE_ID`? You can always override `Authentication` with `fetchOptions`

**Review by Tymek — APPROVED** [review]

File: lib/src/getDefinitions.ts (hunk 2 above)
Comment (Jax-p) [reply lib/src/getDefinitions.ts:14]: > Is there a situation where you'd use both token and UNLEASH_SERVER_INSTANCE_ID?    

I personally don't but based on the documentation I don't think it's inconsistent. 

> You can always override Authentication with fetchOptions

Do you think we would omit the "false" option for the token and let the user overload fetchOptions? In that case it seems less intuitive to me. Besides, it means I would have to set it every time getDefinitions and it wouldn't work globally right?

However, I'll be happy for any variation if it leads to the possibility to omit the token :).

**Comment by Tymek** [comment]
Moved here: https://github.com/Unleash/unleash-client-nextjs/pull/28
Sorry for merge conflicts

**Pull request closed without merging**